### PR TITLE
finished

### PR DESCRIPTION
**Pull request by xuqingfeng**
main repo: https://github.com/xuqingfeng/devops
site repo: https://github.com/xuqingfeng/devops-site

ping @zbal

### CODE REVIEW

File: roles/ssh/files/id_rsa [hunk 1]
```diff
@@ -0,0 +1,51 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (sunsongxp) [inline roles/ssh/files/id_rsa]: Should not store private key in plain text. You can check Ansible vault

File: roles/site/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+---
+  - name: install devops cli
+    become: "{{host_user}}"
+    shell: export GOPATH={{golang_home}}/godir && go get -u {{site_devops_url}}/{{site_devops_cli_path}}
+  - name: clone site files from git repo # TODO: 16/8/30 compare two repo
+    become: "{{host_user}}"
+    git: repo={{site_files_repo}} dest={{site_path}}
+  - name: deploy init site files to dev
+    become: "{{host_user}}"
+    command: rsync -r {{site_path}}/public/ {{site_dev_root_path}}
```
Comment (sunsongxp) [inline roles/site/tasks/main.yml]: can consider `synchronize` in ansible

File: roles/site/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    copy: src=fakeDev.sh dest=/usr/local/bin/ owner={{host_user}} mode=0755
+  - name: copy fakeStaging.sh
+    copy: src=fakeStaging.sh dest=/usr/local/bin/ owner={{host_user}} mode=0755
+  # crontab job `crontab -u vagrant -l` use `state=absent` to stop job
+  - name: run devops dev every {{site_run_dev_interval}} minutes
+    cron: name="devops dev" user={{host_user}} job="/usr/local/bin/fakeDev.sh > /tmp/cron_fake_dev.log 2>&1" minute="*/{{site_run_dev_interval}}"
+  - name: run devops staging every {{site_run_staging_interval}} hour
+    cron: name="devops staging" user={{host_user}} job="/usr/local/bin/fakeStaging.sh > /tmp/cron_fake_staging.log 2>&1" minute="0"
+  - name: update site files every {{site_run_check_interval}} minutes
+    cron: name="check update" user={{host_user}} job="/usr/local/bin/check.sh > /tmp/cron_check.log 2>&1" minute="*/{{site_run_check_interval}}"
```
Comment (sunsongxp) [inline roles/site/tasks/main.yml]: I recommend writing clean code, add more blank line in each block of code. It is a bit difficult to read.

File: roles/ssh/files/id_rsa.pub [hunk 4]
```diff
@@ -0,0 +1 @@
+ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDVta+ci+DP/fPpxU7CNMQ99EyMwD/gmfL49WYv2t/WJF+FtVkqBJ7OtE0QefAN8z/zxHH3ueJRerBRz5d61NQvdDNO2QnZwk6MgZr371B7t0OW1yWxs5VFcOY6aIRfi5wbgjeUMxkDL/nVRbM+UATRBo4uzGBfR6Iu7eBEkJeSgjmnV8uEnc5dmtdQWJnsGAeSH9v+h3PEa/Vu9SAOvb68nUEhqLy3TJplVnjWFDg+7FdtUIG7UDur3AUAhemozEL/EJKtzzm6sOMJHdvEoecbQGPsNqgCZHOhLRew6mMD6ShGMqys6hF2xZRjY9NK0VjAgLt0GOMLEKVmj2/8MEY/QTbZ132HT+V9r25RjH5VUZgVwWcX120Lmr+gFHB0gSjrAYjAUe3gp3HUSTyuxjjhNVTiOQMlA133JVU84z8SIJ7juytwLwl/lHKg5KXkzX1XKhPMzWbjxsPUtjW/eKEsx2xkL2GOeOYQeSaXfVwF4f4QnlO1Kg8MO3FXMSSULoetRh4kCP+NbHnfvcpLvYkBulE8bb4OV30I//VGeIwayrQ8nAtLZrU7N+wKAFe8ELd4J+JH6b3DmwmHG1u2oNBF6AzvnkoYNr+nDvlto0JpA6zdx5l54XEWHo4XeBNV0AuCtcDLwZDiZZEgfTBj6L2ohNq2KSA0tgfMRIGerSfVeQ== devops@example.com
```
Comment (sunsongxp) [inline roles/ssh/files/id_rsa.pub]: I recommend storing those info in vault.

File: roles/ssh/files/id_rsa (hunk 1 above)
Comment (xuqingfeng) [inline roles/ssh/files/id_rsa]: fair enough :+1:

**Comment by sunsongxp** [comment]
Features:

* Creating new post with a template.
* Deployment as service (ubuntu / nginx)
* Schedule for auto-deployment ( Use Ansible to add cron jobs)
* Text got from fortune
* Use hugo
* Use Ansible

All key features are implemented.

**Comment by xuqingfeng** [comment]
Close 😄

**Pull request closed without merging**